packages/neuron-ui/src/components/DetectDuplicateWalletDialog/index.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { useMemo, useState, useCallback } from 'react'
+import { useTranslation } from 'react-i18next'
+import Dialog from 'widgets/Dialog'
+import RadioGroup from 'widgets/RadioGroup'
+import { useState as useGlobalState, useDispatch, AppActions } from 'states'
+import { requestPassword } from 'services/remote'
+import styles from './detectDuplicateWalletDialog.module.scss'
+
+const DetectDuplicateWalletDialog = ({ onClose }: { onClose: () => void }) => {
+  const {
+    wallet: { id: currentID = '' },
+    settings: { wallets = [] },
+  } = useGlobalState()
+  const dispatch = useDispatch()
+  const [deletableWallets, setDeletableWallets] = useState<string[]>([])
+  const [t] = useTranslation()
+
+  const groups = useMemo(() => {
+    const obj: {
+      [key: string]: State.WalletIdentity[]
+    } = {}
+    wallets.forEach(item => {
+      if (item.extendedKey in obj) {
+        obj[item.extendedKey].push(item)
+      } else {
+        obj[item.extendedKey] = [item]
+      }
+    })
+
+    return Object.values(obj).filter(list => list.length > 1)
+  }, [wallets])
+
+  const handleGroupChange = useCallback(
+    (checked: string) => {
+      const [extendedKey, id] = (checked as string).split('_')
+      const list: string[] = []
+      wallets.forEach(item => {
+        if (item.extendedKey === extendedKey) {
+          if (item.id !== id) {
+            list.push(item.id)
+          }
+        } else if (deletableWallets.includes(item.id)) {
+          list.push(item.id)
+        }
+      })
+
+      setDeletableWallets(list)
+    },
+    [wallets, deletableWallets, setDeletableWallets]
+  )
+
+  const onConfirm = useCallback(async () => {
+    const getRequest = (id: string) => {
+      if (wallets.find(item => (item.id === id && item.device) || item.isWatchOnly)) {
+        return requestPassword({ walletID: id, action: 'delete-wallet' })
+      }
+      return new Promise(resolve => {
+        dispatch({
+          type: AppActions.RequestPassword,
+          payload: {
+            walletID: id,
+            actionType: 'delete',
+            onSuccess: async () => {
+              await resolve(id)
+            },
+          },
+        })
+      })
+    }
+
+    let ids = [...deletableWallets]
```

Comment:
any reason for the shallow copy here? maybe the `Set` is helpful here to remove the duplicated ids

```ts
const ids = new Set(deletableWallets)
```

Reply:
There is no need to remove the duplicated ids here. Just need to move the currentID to the last position of ids.

Reply:
I see. But this piece is a bit confusing for me, it would be more understandable if we refactor it like this

```ts
const requestToDeleteIds = duplicatedWallets
  .filter(item => item !== currentID)
  .concat(duplicatedWallets.include(currentID) ? [currentID] : [])
```